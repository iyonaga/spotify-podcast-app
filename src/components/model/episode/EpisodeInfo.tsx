import CalendarIcon from '@/components/icons/Calendar';
import ClockIcon from '@/components/icons/Clock';
import { millisToMinutes } from '@/lib/time';

interface Props {
  episode: SpotifyApi.EpisodeObjectSimplified;
}

const EpisodeInfo: React.VFC<Props> = ({ episode }) => {
  return (
    <div className="flex">
      <div className="flex items-center">
        <CalendarIcon height={15} />
        <p className="relative top-[1px] ml-[7px] text-[14px]">
          {episode.release_date}
        </p>
      </div>
      <div className="flex items-center ml-[15px]">
        <ClockIcon height={15} />
        <p className="relative top-[1px] ml-[7px] text-[14px]">
          {millisToMinutes(episode.duration_ms)}åˆ†
        </p>
      </div>
    </div>
  );
};

export default EpisodeInfo;
