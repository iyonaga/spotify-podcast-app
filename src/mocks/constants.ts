const API_BASE_URL = 'https://api.spotify.com/v1';

export const ENDPOINTS = {
  // shows
  GET_SHOW_EPISODES: `${API_BASE_URL}/shows/:id/episodes`,
  SAVE_SHOWS: `${API_BASE_URL}/me/shows`,
  REMOVE_SAVED_SHOWS: `${API_BASE_URL}/me/shows`,
  CHECK_SAVED_SHOWS: `${API_BASE_URL}/me/shows/contains`,

  // episodes
  GET_SAVED_EPISODES: `${API_BASE_URL}/me/episodes`,
  SAVE_EPISODES: `${API_BASE_URL}/me/episodes`,
  REMOVE_SAVED_EPISODES: `${API_BASE_URL}/me/episodes`,
  CHECK_SAVED_EPISODES: `${API_BASE_URL}/me/episodes/contains`,
};
