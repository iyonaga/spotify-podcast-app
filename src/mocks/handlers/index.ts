import { handlers as episodesHandlers } from './episodes';
import { handlers as searchHandlers } from './search';
import { handlers as showsHandlers } from './shows';

export const handlers = [
  ...showsHandlers,
  ...episodesHandlers,
  ...searchHandlers,
];
