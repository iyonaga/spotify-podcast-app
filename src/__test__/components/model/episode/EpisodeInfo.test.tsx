import { render } from '@testing-library/react';
import EpisodeInfo from '@/components/model/episode/EpisodeInfo';
import { episodesMock } from '@/mocks/episodesMock';

describe('components/model/episode/EpisodeInfo.tsx', () => {
  test('スナップショットが一致する', () => {
    const { asFragment } = render(<EpisodeInfo episode={episodesMock[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
