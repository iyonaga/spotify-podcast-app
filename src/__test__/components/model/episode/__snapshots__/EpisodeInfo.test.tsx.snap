// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/model/episode/EpisodeInfo.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <div
    class="flex"
  >
    <div
      class="flex items-center"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 16 17"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.2237 1C11.2237 0.585786 10.8879 0.25 10.4737 0.25C10.0595 0.25 9.72368 0.585786 9.72368 1H11.2237ZM9.72368 4.15789C9.72368 4.57211 10.0595 4.90789 10.4737 4.90789C10.8879 4.90789 11.2237 4.57211 11.2237 4.15789H9.72368ZM6.92105 1.82895C6.50684 1.82895 6.17105 2.16473 6.17105 2.57895C6.17105 2.99316 6.50684 3.32895 6.92105 3.32895V1.82895ZM4.15789 3.32895C4.57211 3.32895 4.90789 2.99316 4.90789 2.57895C4.90789 2.16473 4.57211 1.82895 4.15789 1.82895V3.32895ZM4.90789 1C4.90789 0.585786 4.57211 0.25 4.15789 0.25C3.74368 0.25 3.40789 0.585786 3.40789 1H4.90789ZM3.40789 4.15789C3.40789 4.57211 3.74368 4.90789 4.15789 4.90789C4.57211 4.90789 4.90789 4.57211 4.90789 4.15789H3.40789ZM13.2368 1.82895C12.8226 1.82895 12.4868 2.16473 12.4868 2.57895C12.4868 2.99316 12.8226 3.32895 13.2368 3.32895V1.82895ZM11.2237 2.57895V1H9.72368V2.57895H11.2237ZM9.72368 2.57895V4.15789H11.2237V2.57895H9.72368ZM10.4737 1.82895H6.92105V3.32895H10.4737V1.82895ZM0.25 7.31579V14.4211H1.75V7.31579H0.25ZM0.25 14.4211C0.25 15.7073 1.29271 16.75 2.57895 16.75V15.25C2.12112 15.25 1.75 14.8789 1.75 14.4211H0.25ZM2.57895 16.75H13.6316V15.25H2.57895V16.75ZM13.6316 16.75C14.9178 16.75 15.9605 15.7073 15.9605 14.4211H14.4605C14.4605 14.8789 14.0894 15.25 13.6316 15.25V16.75ZM15.9605 14.4211V7.31579H14.4605V14.4211H15.9605ZM15.2105 6.56579H1V8.06579H15.2105V6.56579ZM1.75 7.31579V4.15789H0.25V7.31579H1.75ZM1.75 4.15789C1.75 3.70008 2.12113 3.32895 2.57895 3.32895V1.82895C1.2927 1.82895 0.25 2.87165 0.25 4.15789H1.75ZM2.57895 3.32895H4.15789V1.82895H2.57895V3.32895ZM3.40789 1V4.15789H4.90789V1H3.40789ZM15.9605 7.31579V4.15789H14.4605V7.31579H15.9605ZM15.9605 4.15789C15.9605 2.87166 14.9179 1.82895 13.6316 1.82895V3.32895C14.0894 3.32895 14.4605 3.70007 14.4605 4.15789H15.9605ZM13.6316 1.82895H13.2368V3.32895H13.6316V1.82895Z"
          fill="white"
        />
      </svg>
      <p
        class="relative top-[1px] ml-[7px] text-[14px]"
      >
        2022-02-05
      </p>
    </div>
    <div
      class="flex items-center ml-[15px]"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 17 17"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.25 4C9.25 3.58579 8.91421 3.25 8.5 3.25C8.08579 3.25 7.75 3.58579 7.75 4H9.25ZM8.5 8.5H7.75C7.75 8.91421 8.08579 9.25 8.5 9.25V8.5ZM13 9.25C13.4142 9.25 13.75 8.91421 13.75 8.5C13.75 8.08579 13.4142 7.75 13 7.75V9.25ZM7.75 4V8.5H9.25V4H7.75ZM8.5 9.25H13V7.75H8.5V9.25ZM8.5 16.75C13.0563 16.75 16.75 13.0563 16.75 8.5H15.25C15.25 12.2279 12.2279 15.25 8.5 15.25V16.75ZM16.75 8.5C16.75 3.94365 13.0563 0.25 8.5 0.25V1.75C12.2279 1.75 15.25 4.77208 15.25 8.5H16.75ZM8.5 0.25C3.94365 0.25 0.25 3.94365 0.25 8.5H1.75C1.75 4.77208 4.77208 1.75 8.5 1.75V0.25ZM0.25 8.5C0.25 13.0563 3.94365 16.75 8.5 16.75V15.25C4.77208 15.25 1.75 12.2279 1.75 8.5H0.25Z"
          fill="white"
        />
      </svg>
      <p
        class="relative top-[1px] ml-[7px] text-[14px]"
      >
        12分
      </p>
    </div>
  </div>
</DocumentFragment>
`;
